### PR TITLE
ARROW-9109: [Python][Packaging] Enable S3 support in manylinux wheels

### PR DESCRIPTION
**Pull request by pitrou**

### CODE REVIEW

**Comment by pitrou** [comment]
@github-actions crossbow submit -g wheel

**Comment by pitrou** [comment]
@github-actions crossbow submit -g wheel

**Comment by pitrou** [comment]
@github-actions crossbow submit -g wheel

**Comment by pitrou** [comment]
It seems like everything is working fine here.

**Comment by pitrou** [comment]
This grows the manylinux wheel sizes by about 2-3 MB (from ~13MB to ~15-16MB). It think it's reasonable.

**Comment by kszucs** [comment]
> This grows the manylinux wheel sizes by about 2-3 MB (from ~13MB to ~15-16MB). It think it's reasonable.

Yes, I expected more.

**Comment by wesm** [comment]
Very reasonable. Thanks @pitrou!

**Comment by kszucs** [comment]
Quickly went through it and LGTM. I can give it a more thorough review tomorrow.

File: python/manylinux1/scripts/build_boost.sh [hunk 1]
```diff
@@ -20,10 +20,16 @@ BOOST_VERSION=1.73.0
...
 NCORES=$(($(grep -c ^processor /proc/cpuinfo) + 1))
 
-curl -sL https://sourceforge.net/projects/boost/files/boost/${BOOST_VERSION}/boost_${BOOST_VERSION_UNDERSCORE}.tar.bz2 -o /boost_${BOOST_VERSION_UNDERSCORE}.tar.bz2
-tar xf boost_${BOOST_VERSION_UNDERSCORE}.tar.bz2
+BASE_NAME=boost_${BOOST_VERSION_UNDERSCORE}
+ARCHIVE_NAME=boost_${BOOST_VERSION_UNDERSCORE}.tar.bz2
+
+# Bintray is much faster but can fail because of limitations
+curl -sL https://dl.bintray.com/boostorg/release/${BOOST_VERSION}/source/boost_${BOOST_VERSION_UNDERSCORE}.tar.bz2 -o /${ARCHIVE_NAME} \
+    || curl -sL https://sourceforge.net/projects/boost/files/boost/${BOOST_VERSION}/boost_${BOOST_VERSION_UNDERSCORE}.tar.bz2 -o /${ARCHIVE_NAME}
```
Comment (kszucs) [inline python/manylinux1/scripts/build_boost.sh:28]: Nice!

**Review by kszucs — APPROVED** [review]
The OSX failure is unrelated.

File: python/manylinux1/scripts/build_boost.sh (hunk 1 above)
Comment (kou) [reply python/manylinux1/scripts/build_boost.sh:28]: FYI: I've created https://bintray.com/apache/arrow/boost-mirror and mirrored Boost 1.73.0 as https://apache.bintray.com/arrow/boost/1.73.0/boost_1_73_0.tar.bz2 .

We can use this to save the official Boost download resource.

**Pull request closed without merging**